haystack/nodes/retriever/_base_encoder.py

```diff
@@ -0,0 +1,102 @@
+import logging
+from abc import abstractmethod
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import numpy as np
+
+from haystack.document_stores.base import BaseDocumentStore
+from haystack.schema import Document
+
```

Comment:
Could we consider renaming this file to `_base_embedding_encoder.py` to make it clear it is related to the `_embedding_encoder.py` file?